tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -961,3 +968,309 @@ def progress_check():
+                else:
+                    # Extract data
+                    timings = data.get('timings', {})
+                    log_timings_with_percentiles(timings)
+            else:
+                data = {}
+
+        topic_details = data.get('topics', [])
+        current_count = len(topic_details)
+        self.logger.info(f"Created {current_count} topics")
+        assert len(topic_details) == topic_count, \
+            f"Topic count not reached: {current_count}/{topic_count}"
+
+        # Validate topics
+        topics = self.rpk.list_topics(detailed=True)
+
+        # Validate topic creation
+        self.logger.info("Validating created topics")
+        topics_ok = []
+        topics_failed = []
+
+        # Searches for a topic in our list
+        def get_topic_index(name):
+            for idx in range(len(topic_details)):
+                if name == topic_details[idx]['name']:
+                    return idx
+            return -1
+
+        # Validate RP topics against requested details
+        for name, partitions, replicas in topics:
+            err = "ok"
+            idx = get_topic_index(name)
+            if idx < 0:
+                # RP has topic that is not ours
+                # Cluster was clean and it is an issue
+                err = f"Unexpected topic found: {name}"
+            elif int(replicas) != topic_details[idx]['replicas']:
+                # Replication factor is wrong
+                err = f"Replication factor error: current {replicas}, " \
+                f"target {topic_details[idx]['replicas']}"
+            elif int(partitions) != topic_details[idx]['partitions']:
+                # Partitions number is wrong
+                err = f"Partitions number error: current {partitions}, " \
+                f"target {topic_details[idx]['partitions']}"
+            else:
+                # Topic passed the check
+                topics_ok.append(topic_details[idx]['name'])
+                continue
+            # one of errors happened, add topic to failed list
+            # And add error message
+            failed_topic = (topic_details[idx], err)
+            self.logger.warning(f"Topic {topic_details[idx]['name']}, {err}")
+            topics_failed.append(failed_topic)
+
+        # Check that no errors happened
+        failed = len(topics_failed)
+        assert failed < 1, f"{failed} RP topics has errors"
+
+        topics_available = []
+        topics_unavailable = []
+        # Validate our list against RP
+        for topic in topic_details:
+            # make sure topic is present in RP
+            if topic['name'] not in topics_ok:
+                topics_unavailable.append(topic['name'])
+            else:
+                topics_available.append(topic['name'])
+        unavailable = len(topics_unavailable)
+        assert unavailable < 1, f"{unavailable} topics are missing in RP"
+
+        # Free node that used to create topics
+        self.cluster.free_single(node)
```

Comment:
Why do we allocate an entire node for creating the topics? It seems like the ducktape driver node can create them just as well.

Reply:
So that we do not be limited by the lack of clien's node resources and able to send requests as fast as possible. Client node has significantly less resources comparing to worker ones. When checking the 40k - it would be more significant.